src/golog_app/app.cpp

```diff
@@ -0,0 +1,281 @@
+	;
+    boost::program_options::variables_map variables;
+	boost::program_options::store(boost::program_options::parse_command_line(argc, argv, options),
+	                              variables);
+
+	for (const auto& it : variables) {
+		std::cout << it.first.c_str() << " ";
+		auto& value = it.second.value();
+		if (auto v = boost::any_cast<uint32_t>(&value))
+			std::cout << *v << std::endl;
+		else if (auto v = boost::any_cast<std::string>(&value))
+			std::cout << *v << std::endl;
+		else if (auto v = boost::any_cast<std::filesystem::path>(&value))
+			std::cout << *v << std::endl;
+		else
+			std::cout << "error" << std::endl;
+	}
+
+	if (variables.count("help")) {
+		SPDLOG_INFO(options);
+		show_help = true;
+		return;
+	}
+	boost::program_options::notify(variables);
+	debug                  = variables["debug"].as<bool>();
+	hide_controller_labels = variables["hide-controller-labels"].as<bool>();
+
+	// Convert the vector of actions into a set of actions.
+	if (variables.count("controller-action")) {
+		std::copy(std::begin(variables["controller-action"].as<std::vector<std::string>>()),
+		          std::end(variables["controller-action"].as<std::vector<std::string>>()),
+		          std::inserter(controller_actions, std::end(controller_actions)));
+	}
+	// Convert the vector of actions into a set of actions.
+	if (variables.count("environment-action")) {
+		std::copy(std::begin(variables["environment-action"].as<std::vector<std::string>>()),
+		          std::end(variables["environment-action"].as<std::vector<std::string>>()),
+		          std::inserter(environment_actions, std::end(environment_actions)));
+	}
+}
+
+void
+read_proto_from_file(const std::filesystem::path &path, google::protobuf::Message *output)
+{
+	int fd = open(path.c_str(), O_RDONLY);
+	if (fd < 0) {
+		throw std::invalid_argument(
+		  fmt::format("Could not open spec file '{}' (errno: {})", path.c_str(), errno));
+	}
+	google::protobuf::io::FileInputStream stream(fd);
+	stream.SetCloseOnDelete(true);
+	if (!google::protobuf::TextFormat::Parse(&stream, output)) {
+		throw std::invalid_argument(fmt::format("Failed to read proto from file '{}", path.c_str()));
+	}
+}
+
+GologProgram
+read_golog_from_file(const std::filesystem::path &path, std::set<std::string> relevant_fluents)
+{
+	std::ifstream f(path.c_str());
+	if(!f) {
+		throw std::invalid_argument(
+		  fmt::format("Could not open golog file '{}' (errno: {})", path.c_str(), errno));
+	}
+	std::ostringstream ss;
+	ss << f.rdbuf();
+
+	return GologProgram(ss.str(), relevant_fluents);
+}
+
+std::set<std::string>
+unwrap_fluents(std::set<logic::AtomicProposition<std::set<std::string>>> input)
+{
+	std::set<std::string> res;
+	for (const auto &i : input) {
+		for (const auto &s : i.ap_) {
+			res.insert(s);
+		}
+	}
+	return res;
+}
+
+void
+Launcher::run()
+{
+	if (show_help) {
+		return;
+	}
+
+	SPDLOG_INFO("Using maximum clock constraint K={}", K);
```

Comment:
`K` is not the maximum clock constraint, but the maximal constant appearing in any transition or constraint